qiskit/transpiler/passes/scheduling/calibration_creators.py

```diff
@@ -0,0 +1,199 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Calibration creators."""
+
+import math
+from typing import List
+from abc import ABC, abstractmethod
+import numpy as np
+
+from qiskit.pulse import Play, ShiftPhase, Schedule, ControlChannel, DriveChannel, GaussianSquare
+from qiskit.exceptions import QiskitError
+from qiskit.providers import basebackend
+from qiskit.dagcircuit import DAGNode
+from qiskit.circuit.library.standard_gates import RZXGate
+
+
+class CalibrationCreator(ABC):
+    """Abstract base class to inject calibrations into circuits."""
+
+    @abstractmethod
+    def supported(self, node_op: DAGNode) -> bool:
+        """Determine if a given name supports the calibration."""
+
+    @abstractmethod
+    def get_calibration(self, params: List, qubits: List) -> Schedule:
+        """Gets the calibrated schedule for the given qubits and parameters."""
+
+
+class ZXScheduleBuilder(CalibrationCreator):
```

Comment:
maybe just call this `RZXCalibrationBuilder` so it's clear that it's of that class.

Reply:
Done in 966f484